libraries/AP_InertialSensor/AP_InertialSensor_Backend.h

```diff
@@ -232,15 +232,29 @@ class AP_InertialSensor_Backend
     // return the requested sample rate in Hz
     uint16_t get_sample_rate_hz(void) const;
 
+    // return the notch filter center in Hz for the sample rate
+    uint16_t _gyro_notch_center_freq_hz(void) const { return _imu._notch_filter.center_freq_hz(); }
+
+    // return the notch filter bandwidth in Hz for the sample rate
+    uint8_t _gyro_notch_bandwidth_hz(void) const { return _imu._notch_filter.bandwidth_hz(); }
+
+    // return the notch filter attenuation in dB for the sample rate
+    float _gyro_notch_attenuation_dB(void) const { return _imu._notch_filter.attenuation_dB(); }
+
+    uint8_t _gyro_notch_enabled(void) const { return _imu._notch_filter.enabled(); }
+
     // common gyro update function for all backends
     void update_gyro(uint8_t instance);
 
     // common accel update function for all backends
     void update_accel(uint8_t instance);
 
     // support for updating filter at runtime
-    int8_t _last_accel_filter_hz[INS_MAX_INSTANCES];
-    int8_t _last_gyro_filter_hz[INS_MAX_INSTANCES];
+    uint8_t _last_accel_filter_hz[INS_MAX_INSTANCES];
```

Comment:
given we now allow filtering at higher frequencies I think these should be uint16_t to allow for more than 256Hz filter freq

Reply:
Definitely! I've already tested this.

I think a similar change can be done also in **NotchFilterParams** on `bandwidth_hz`, at least with an uint8, I'm using a 240hz bandwidth and it works very well!

Reply:
Yes, agreed - I wasn't sure whether there was some constraint on using more layout storage or changing parameter types (AP_Int8 -> APInt16). I assume not? Will fix.